src/components/AppBreadcrumbs/AppBreadcrumbs.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import { Breadcrumbs, Link } from '@material-ui/core';
+import { Link as RouterLink } from 'react-router-dom';
+
+function capitalize(str: string): string {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+const styles = (theme: any) => ({
+  terminalBreadcrumb: {
+    color: theme.palette.primary.dark,
+    cursor: 'default',
+    '&:hover': {
+      textDecoration: 'none',
+    },
+  },
+});
+
+export type Context = { type: string; id: string; name: string };
+export type Breadcrumb = { text: string; to: string };
+
+type AppBreadcrumbsProps = {
+  childBreadcrumbs: Array<Breadcrumb>;
+  context: Context;
+};
+
+const BreadcrumbLink = withStyles(styles)((props: any) => {
+  const { classes, ...other } = props;
+  if (props.disabled) {
+    return (
+      <Link
+        component={RouterLink}
+        color="primary"
+        className={classes.terminalBreadcrumb}
+        onClick={(e) => e.preventDefault()}
+        {...other}
+      />
+    );
+  }
+  return <Link component={RouterLink} color="primary" {...other} />;
+});
+
+const AppBreadcrumbs: React.FC<AppBreadcrumbsProps> = (props) => {
+  const { context, childBreadcrumbs } = props;
```

Comment:
It's been forever since I've looked at typescript -- This could still be a functional component, right?
```suggestion
function AppBreadcrumbs: React.FC<AppBreadcrumbsProps> = ({
  context, childBreadcrumbs
  }) => { 
```

Reply:
```suggestion
const AppBreadcrumbs = ({context, childBreadcrumbs}: AppBreadcrumbsProps) => {
```
Ok, does this work? Following the example[ here](https://github.com/typescript-cheatsheets/react#function-components) - Seems that React.FC is discouraged.